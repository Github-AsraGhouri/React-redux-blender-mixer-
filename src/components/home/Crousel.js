import React, { Component } from 'react';

class Crousel extends Component {
    constructor(props) {
        super(props)
        this.state = {
            selectedImg: this.props.PrimaryImage
        }
    }

    changeImage = (img) => {
        this.setState({ selectedImg: img.image })
    }

    render() {
        return (
            <div className="container">
                <p className="Blender-details">
                    {this.props.title}
                </p>
                <div className="mixer-blender">
                    <img className="d-block w-100" src={this.state.selectedImg} alt="Second slide" />
                </div>
                <p className="view-larger">
                    <img src={require('./../../assets/img/plus-zoom.png')} alt="plus-zoom" width="30px;" />
                    view larger
                  </p>
                <div className="container-fluid">
                    <div id="carouselExample" className="carousel slide" data-ride="carousel" data-interval="9000">
                        <div className="carousel-inner row w-100 mx-auto" role="listbox">
                            {
                                this.props.AlternateImages && this.props.AlternateImages.map((img, i) => {
                                    return (
                                        <div onClick={() => this.changeImage(img)} className={i === 0 ? "carousel-item col-md-3 active" : "carousel-item col-md-3"}>
                                            <div className="panel panel-default">
                                                <div className="panel-thumbnail">
                                                    <div className="thumb mouse-pointer">
                                                        <img className="img-fluid mx-auto d-block" src={img.image} alt="slide 1" />
                                                    </div>
                                                </div>
                                            </div>
                                        </div>
                                    )
                                })
                            }
                        </div>
                        <a className="carousel-control-prev" href="#carouselExample" role="button" data-slide="prev">
                            <i class="fa fa-angle-left custom" ></i>
                            <span className="sr-only">Previous</span>
                        </a>
                        <a className="carousel-control-next" href="#carouselExample" role="button" data-slide="next">
                            <i class="fa fa-angle-right custom" ></i>
                            <span className="sr-only">Next</span>
                        </a>
                    </div>
                </div>
            </div>
        );
    }
}

export default Crousel;