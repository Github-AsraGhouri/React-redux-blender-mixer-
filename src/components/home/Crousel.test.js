import React from 'react';
import { shallow } from 'enzyme';
import Crousel from './Crousel';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

describe('<Crousel />', () => {
    it('renders component with props', () => {
      const component = shallow(<Crousel props/>);
      expect(component).toMatchSnapshot();
    });
    
    it('renders component HTML', () => {
      const componentHtml = shallow(<Crousel />);
      expect(componentHtml.find('div').length).toEqual(5);
      expect(componentHtml.find('p').length).toEqual(2);
      expect(componentHtml.find('span').length).toEqual(2);
      expect(componentHtml.find('img').length).toEqual(2);
      expect(componentHtml.find('i').length).toEqual(2);
    });
  });