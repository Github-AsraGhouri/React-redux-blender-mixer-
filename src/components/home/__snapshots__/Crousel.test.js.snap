// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Crousel /> renders component with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Crousel
    props={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="Blender-details"
        />,
        <div
          className="mixer-blender"
        >
          <img
            alt="Second slide"
            className="d-block w-100"
          />
        </div>,
        <p
          className="view-larger"
        >
          <img
            alt="plus-zoom"
            src="plus-zoom.png"
            width="30px;"
          />
          view larger
        </p>,
        <div
          className="container-fluid"
        >
          <div
            className="carousel slide"
            data-interval="9000"
            data-ride="carousel"
            id="carouselExample"
          >
            <div
              className="carousel-inner row w-100 mx-auto"
              role="listbox"
            />
            <a
              className="carousel-control-prev"
              data-slide="prev"
              href="#carouselExample"
              role="button"
            >
              <i
                class="fa fa-angle-left custom"
              />
              <span
                className="sr-only"
              >
                Previous
              </span>
            </a>
            <a
              className="carousel-control-next"
              data-slide="next"
              href="#carouselExample"
              role="button"
            >
              <i
                class="fa fa-angle-right custom"
              />
              <span
                className="sr-only"
              >
                Next
              </span>
            </a>
          </div>
        </div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "Blender-details",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Second slide"
            className="d-block w-100"
          />,
          "className": "mixer-blender",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Second slide",
            "className": "d-block w-100",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="plus-zoom"
              src="plus-zoom.png"
              width="30px;"
            />,
            "view larger",
          ],
          "className": "view-larger",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "plus-zoom",
              "src": "plus-zoom.png",
              "width": "30px;",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "view larger",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="carousel slide"
            data-interval="9000"
            data-ride="carousel"
            id="carouselExample"
          >
            <div
              className="carousel-inner row w-100 mx-auto"
              role="listbox"
            />
            <a
              className="carousel-control-prev"
              data-slide="prev"
              href="#carouselExample"
              role="button"
            >
              <i
                class="fa fa-angle-left custom"
              />
              <span
                className="sr-only"
              >
                Previous
              </span>
            </a>
            <a
              className="carousel-control-next"
              data-slide="next"
              href="#carouselExample"
              role="button"
            >
              <i
                class="fa fa-angle-right custom"
              />
              <span
                className="sr-only"
              >
                Next
              </span>
            </a>
          </div>,
          "className": "container-fluid",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="carousel-inner row w-100 mx-auto"
                role="listbox"
              />,
              <a
                className="carousel-control-prev"
                data-slide="prev"
                href="#carouselExample"
                role="button"
              >
                <i
                  class="fa fa-angle-left custom"
                />
                <span
                  className="sr-only"
                >
                  Previous
                </span>
              </a>,
              <a
                className="carousel-control-next"
                data-slide="next"
                href="#carouselExample"
                role="button"
              >
                <i
                  class="fa fa-angle-right custom"
                />
                <span
                  className="sr-only"
                >
                  Next
                </span>
              </a>,
            ],
            "className": "carousel slide",
            "data-interval": "9000",
            "data-ride": "carousel",
            "id": "carouselExample",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "carousel-inner row w-100 mx-auto",
                "role": "listbox",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    class="fa fa-angle-left custom"
                  />,
                  <span
                    className="sr-only"
                  >
                    Previous
                  </span>,
                ],
                "className": "carousel-control-prev",
                "data-slide": "prev",
                "href": "#carouselExample",
                "role": "button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "fa fa-angle-left custom",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Previous",
                    "className": "sr-only",
                  },
                  "ref": null,
                  "rendered": "Previous",
                  "type": "span",
                },
              ],
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    class="fa fa-angle-right custom"
                  />,
                  <span
                    className="sr-only"
                  >
                    Next
                  </span>,
                ],
                "className": "carousel-control-next",
                "data-slide": "next",
                "href": "#carouselExample",
                "role": "button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "fa fa-angle-right custom",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Next",
                    "className": "sr-only",
                  },
                  "ref": null,
                  "rendered": "Next",
                  "type": "span",
                },
              ],
              "type": "a",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="Blender-details"
          />,
          <div
            className="mixer-blender"
          >
            <img
              alt="Second slide"
              className="d-block w-100"
            />
          </div>,
          <p
            className="view-larger"
          >
            <img
              alt="plus-zoom"
              src="plus-zoom.png"
              width="30px;"
            />
            view larger
          </p>,
          <div
            className="container-fluid"
          >
            <div
              className="carousel slide"
              data-interval="9000"
              data-ride="carousel"
              id="carouselExample"
            >
              <div
                className="carousel-inner row w-100 mx-auto"
                role="listbox"
              />
              <a
                className="carousel-control-prev"
                data-slide="prev"
                href="#carouselExample"
                role="button"
              >
                <i
                  class="fa fa-angle-left custom"
                />
                <span
                  className="sr-only"
                >
                  Previous
                </span>
              </a>
              <a
                className="carousel-control-next"
                data-slide="next"
                href="#carouselExample"
                role="button"
              >
                <i
                  class="fa fa-angle-right custom"
                />
                <span
                  className="sr-only"
                >
                  Next
                </span>
              </a>
            </div>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Blender-details",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Second slide"
              className="d-block w-100"
            />,
            "className": "mixer-blender",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Second slide",
              "className": "d-block w-100",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="plus-zoom"
                src="plus-zoom.png"
                width="30px;"
              />,
              "view larger",
            ],
            "className": "view-larger",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "plus-zoom",
                "src": "plus-zoom.png",
                "width": "30px;",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "view larger",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="carousel slide"
              data-interval="9000"
              data-ride="carousel"
              id="carouselExample"
            >
              <div
                className="carousel-inner row w-100 mx-auto"
                role="listbox"
              />
              <a
                className="carousel-control-prev"
                data-slide="prev"
                href="#carouselExample"
                role="button"
              >
                <i
                  class="fa fa-angle-left custom"
                />
                <span
                  className="sr-only"
                >
                  Previous
                </span>
              </a>
              <a
                className="carousel-control-next"
                data-slide="next"
                href="#carouselExample"
                role="button"
              >
                <i
                  class="fa fa-angle-right custom"
                />
                <span
                  className="sr-only"
                >
                  Next
                </span>
              </a>
            </div>,
            "className": "container-fluid",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="carousel-inner row w-100 mx-auto"
                  role="listbox"
                />,
                <a
                  className="carousel-control-prev"
                  data-slide="prev"
                  href="#carouselExample"
                  role="button"
                >
                  <i
                    class="fa fa-angle-left custom"
                  />
                  <span
                    className="sr-only"
                  >
                    Previous
                  </span>
                </a>,
                <a
                  className="carousel-control-next"
                  data-slide="next"
                  href="#carouselExample"
                  role="button"
                >
                  <i
                    class="fa fa-angle-right custom"
                  />
                  <span
                    className="sr-only"
                  >
                    Next
                  </span>
                </a>,
              ],
              "className": "carousel slide",
              "data-interval": "9000",
              "data-ride": "carousel",
              "id": "carouselExample",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "carousel-inner row w-100 mx-auto",
                  "role": "listbox",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      class="fa fa-angle-left custom"
                    />,
                    <span
                      className="sr-only"
                    >
                      Previous
                    </span>,
                  ],
                  "className": "carousel-control-prev",
                  "data-slide": "prev",
                  "href": "#carouselExample",
                  "role": "button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "fa fa-angle-left custom",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Previous",
                      "className": "sr-only",
                    },
                    "ref": null,
                    "rendered": "Previous",
                    "type": "span",
                  },
                ],
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      class="fa fa-angle-right custom"
                    />,
                    <span
                      className="sr-only"
                    >
                      Next
                    </span>,
                  ],
                  "className": "carousel-control-next",
                  "data-slide": "next",
                  "href": "#carouselExample",
                  "role": "button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "fa fa-angle-right custom",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Next",
                      "className": "sr-only",
                    },
                    "ref": null,
                    "rendered": "Next",
                    "type": "span",
                  },
                ],
                "type": "a",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
